test/e2e/service.go

```diff
@@ -246,6 +249,94 @@ var _ = Describe("Services", func() {
 		}()
 	}, 240.0)
 
+	It("should be able to create a functioning external load balancer", func() {
+		serviceName := "external-lb-test"
+		ns := api.NamespaceDefault
+		labels := map[string]string{
+			"key0": "value0",
+		}
+		service := &api.Service{
+			ObjectMeta: api.ObjectMeta{
+				Name: serviceName,
+			},
+			Spec: api.ServiceSpec{
+				Port:                       80,
+				Selector:                   labels,
+				TargetPort:                 util.NewIntOrStringFromInt(80),
+				CreateExternalLoadBalancer: true,
+			},
+		}
+
+		By("cleaning up previous service " + serviceName + " from namespace " + ns)
+		c.Services(ns).Delete(serviceName)
+
+		By("creating service " + serviceName + " with external load balancer in namespace " + ns)
+		result, err := c.Services(ns).Create(service)
+		Expect(err).NotTo(HaveOccurred())
+		defer func(ns, serviceName string) { // clean up when we're done
+			By("deleting service " + serviceName + " in namespace " + ns)
+			err := c.Services(ns).Delete(serviceName)
+			Expect(err).NotTo(HaveOccurred())
+		}(ns, serviceName)
+
+		if len(result.Spec.PublicIPs) != 1 {
+			Failf("got unexpected number (%d) of public IPs for externally load balanced service: %v", result.Spec.PublicIPs, result)
+		}
+		ip := result.Spec.PublicIPs[0]
+		port := result.Spec.Port
+
+		pod := &api.Pod{
+			TypeMeta: api.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1beta1",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name:   "elb-test-" + string(util.NewUUID()),
+				Labels: labels,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "webserver",
+						Image: "kubernetes/test-webserver",
+					},
+				},
+			},
+		}
+
+		By("creating pod to be part of service " + serviceName)
+		podClient := c.Pods(api.NamespaceDefault)
+		defer func() {
```

Comment:
Regarding deletion, probably not a problem, but you could alternatively defer the delete only if you successfully created.

Regarding pod.Name empty, in the code above, you initialize Pod.ObjectMeta.Name but not Pod.Name.  So until some other code sets pod.Name to something other than "", that's what you'll have.  I assume that  podClient.Create(pod) further down in line 314 might set pod.Name to what you want, but does it do it if the call fails?  Not sure.